movement/monge_ampere.py

```diff
@@ -138,7 +139,11 @@ def __init__(self, mesh, monitor_function, **kwargs):
         self.maxiter = kwargs.pop("maxiter", 1000)
         self.rtol = kwargs.pop("rtol", 1.0e-08)
         self.dtol = kwargs.pop("dtol", 2.0)
-        self.fix_boundary_nodes = kwargs.pop("fix_boundary_nodes", False)
+        self.fixed_boundary_segments = kwargs.pop("fixed_boundary_segments", [])
+        if self.fixed_boundary_segments == "on_boundary":
+            self.fixed_boundary_segments = mesh.exterior_facets.unique_markers
+        elif not isinstance(self.fixed_boundary_segments, Iterable):
+            self.fixed_boundary_segments = [self.fixed_boundary_segments]
```

Comment:
If we specify a boundary segment which doesn't exist we get an unhelpful error `ValueError: Cannot fix boundary nodes for periodic segments.`. Could we add a check here that the user provided something sensible?

Reply:
Good point. Fixed in 9feeef7713e32c78aaa3c980aa149e028c262b68.